archivessnake/edit_notes.py

```diff
@@ -7,64 +7,67 @@
+                if note['type'] == args.note_type:
+                    for subnote in note.get('subnotes'):
+                        content = subnote['content']
                         if contains_match(content, args.search_string):
-                            if args.action == "delete":
-                                del notes[idx]
-                                print("{} note deleted from object {}".format(
-                                    args.note_type, record.uri))
-                            if args.action == "modify":
-                                print("{} note was originally {} and was changed to {} in object {}".format(
-                                    args.note_type, content, args.replace_string, record.uri))
-                                subnote["content"] = args.replace_string
-                            log_to_spreadsheet(record)
-                            updated = True
-            if updated:
-                save_record(record.uri, aojson)
+                            handle_matching_notes(args, notes, idx, record, content, subnote, client, resource_id, writer, updated)
+                            
+def handle_matching_notes(args, notes, idx, record, content, subnote, client, resource_id, writer, updated):
+    if args.action == "delete":
+        del notes[idx]
+        print("{} note {} deleted from object {}".format(
+            args.note_type, args.search_string, record['uri']))
+    if args.action == "modify":
+        print("{} note was originally {} and was changed to {} in object {}".format(
+            args.note_type, content, args.replace_string, record['uri']))
+        subnote['content'] = args.replace_string
+    log_to_spreadsheet(record, writer, resource_id, client)
+    updated = True
+    if updated:
+        save_record(client, record['uri'], record)
 
 def contains_match(content, search_string):
     """Returns True if user-provided note input matches the corresponding note within a given ratio (CONFIDENCE_RATIO)."""
     ratio = fuzz.token_sort_ratio(content.lower(), search_string.lower())
     return True if ratio > CONFIDENCE_RATIO else False
 
-
-def save_record(uri, data):
+def save_record(client, uri, data):
     """Posts modifications/deletions to ArchivesSpace"""
-    updated = aspace.client.post(uri, json=data)
+    updated = client.post(uri, json=data)
     updated.raise_for_status()
 
-
-def log_to_spreadsheet(archival_object):
+def log_to_spreadsheet(record, writer, resource_id, client):
     """Logs top container identifier information of changed archival objects to spreadsheet"""
-    for instance in archival_object.instances:
-        top_container = instance.sub_container.top_container
-        container = "{} {}".format(
-            top_container.type.capitalize(), top_container.indicator)
-        writer.writerow([container, archival_object.uri])
+    if record['instances']:
+        for instance in record['instances']:
```

Comment:
```suggestion
    for instance in record.get('instances', []):
```
As above, you can make this a bit more concise and avoid KeyErrors by using `get` with a second argument.